src/index.js

```diff
@@ -345,36 +345,47 @@ export class ULabel {
+                current_id_button.removeClass("sel");
+                let old_id = parseInt(current_id_button.attr("id").split("_").at(-1));
                 tgt_jq.addClass("sel");
                 tgt_jq.removeAttr("href");
                 let idarr = tgt_jq.attr("id").split("_");
                 let rawid = parseInt(idarr[idarr.length - 1]);
                 ul.set_id_dialog_payload_nopin(current_subtask["class_ids"].indexOf(rawid), 1.0);
                 ul.update_id_dialog_display();
 
-                // Get the active annotation, if any
-                let target_id = null;
-                if (current_subtask.state.active_id !== null) {
-                    target_id = JSON.parse(JSON.stringify(current_subtask.state.active_id));
-                } else if (current_subtask.state.move_candidate !== null) {
-                    target_id = JSON.parse(JSON.stringify(current_subtask.state.move_candidate["annid"]));
+                // Update the class of the active annotation, except when toggling on the delete class
+                if (rawid !== DELETE_CLASS_ID) {
+                    // Get the active annotation, if any
+                    let target_id = null;
+                    if (current_subtask.state.active_id !== null) {
+                        target_id = JSON.parse(JSON.stringify(current_subtask.state.active_id));
+                    } else if (current_subtask.state.move_candidate !== null) {
+                        target_id = JSON.parse(JSON.stringify(current_subtask.state.move_candidate["annid"]));
+                    }
+
+                    // Update the class of the active annotation
+                    if (target_id !== null) {
+                        // Set the annotation's class to the selected class
+                        ul.handle_id_dialog_click(ul.state["last_move"], target_id, ul.get_active_class_id_idx());
+                    } else {
+                        // If there is not active annotation, still update the brush circle if in brush mode
+                        ul.recolor_brush_circle();
+                    }
                 }
 
-                // Update the class of the active annotation
-                if (target_id !== null) {
-                    // Set the annotation's class to the selected class
-                    ul.handle_id_dialog_click(ul.state["last_move"], target_id, ul.get_active_class_id_idx());
-                } else {
-                    // If there is not active annotation, still update the brush circle if in brush mode
-                    ul.recolor_brush_circle();
+                // If toggling off a delete class while still in delete mode, re-toggle the delete class
+                // This occurs when using a keybind to change a hovered annotation's class while in delete mode
+                if (old_id === DELETE_CLASS_ID && DELETE_MODES.includes(current_subtask.state.annotation_mode)) {
+                    $("#toolbox_sel_" + DELETE_CLASS_ID).trigger("click");
```

Comment:
Idk but this seems really hacky to me. Is it possible to prevent the keybind from changing the hovered annotation's class while in delete mode?

Reply:
I considered changing the behavior so that keybinds don't work in delete modes, but I think that would be unexpected for the user. The real source of the hackiness is that we're using the `click` event to handle the class id change logic, when really it should be something handled by ulabel's state. A refactor that significant is outside the scope of this fix tho, so I feel fine doing it like this for now.

Reply:
Works for me :+1: